Dockerfile

```diff
@@ -12,6 +12,15 @@ RUN git clone http://git.ipxe.org/ipxe.git && \
       sed -i -e "s/#undef.*NET_PROTO_IPV6/#define NET_PROTO_IPV6/g" config/general.h && \
       make bin/undionly.kpxe bin-x86_64-efi/ipxe.efi bin-x86_64-efi/snponly.efi
 
+RUN dd bs=1024 count=2880 if=/dev/zero of=esp.img && \
+      mkfs.msdos -F 12 -n 'ESP_IMAGE' ./esp.img && \
+      mmd -i esp.img EFI && \
+      mmd -i esp.img EFI/BOOT && \
+      mcopy -i esp.img -v /boot/efi/EFI/BOOT/BOOTX64.EFI ::EFI/BOOT && \
```

Comment:
An alternative could be to assemble the boot loader by hand out of the bits we actually need:

```
grub-mkimage \
            -C xz \
            -O x86_64-efi \
            -p /boot/grub \
            -o $efiboot_mount/efi/boot/bootx64.efi \
            boot linux linuxefi search normal configfile \
            part_gpt btrfs ext2 fat iso9660 loopback \
            test keystatus gfxmenu regexp probe \
            efi_gop efi_uga all_video gfxterm font \
            echo read ls cat png jpeg halt reboot
```

Stolen from [devstack](https://github.com/openstack/ironic/blob/master/devstack/lib/ironic#L2572).

Reply:
Not really, because that wouldn't be able to be signed or it wouldn't be a signed element if we had to do secure boot all along the chain. Then again, it might not be possible, but we're going to have to test it. Part of the reason I pulled in the shim is to ensure that.